src/services/userService.js

```diff
@@ -0,0 +1,45 @@
+let users = [];
+
+const resetUsers = () => {
+  users = [];
+};
+
+const getAllUsers = () => {
+  return users;
+};
+
+const getUserById = (id) => {
+  return users.find((person) => person.id === +id) || null;
+};
+
+const createUser = (name) => {
+  const user = {
+    id: Math.floor(Math.random() * 1000),
+    name,
+  };
+
+  users.push(user);
+
+  return user;
+};
+
+const updateUser = ({ id, name }) => {
+  const user = getUserById(+id);
+
+  Object.assign(user, { name });
+
+  return user;
+};
+
+const deleteUser = (id) => {
+  users = users.filter((person) => person.id !== +id);
+};
```

Comment:
The `deleteUser` function does not provide any feedback about whether the operation was successful or not. If a user with the provided `id` does not exist, the function will silently fail. It's a good practice to return a value indicating the result of the operation, for example, a boolean value or the deleted user object.